stubs/docutils/docutils/parsers/rst/roles.pyi

```diff
@@ -1,20 +1,131 @@
-from _typeshed import Incomplete
 from collections.abc import Callable
 from typing import Any
 from typing_extensions import TypeAlias
 
-import docutils.nodes
 import docutils.parsers.rst.states
+from docutils import nodes
 from docutils.languages import _LanguageModule
+from docutils.nodes import Node, TextElement
+from docutils.parsers.rst.states import Inliner
 from docutils.utils import Reporter, SystemMessage
 
+DEFAULT_INTERPRETED_ROLE: str
+
 _RoleFn: TypeAlias = Callable[
     [str, str, str, int, docutils.parsers.rst.states.Inliner, dict[str, Any], list[str]],
-    tuple[list[docutils.nodes.reference], list[docutils.nodes.reference]],
+    tuple[list[nodes.reference], list[nodes.reference]],
 ]
 
+def register_canonical_role(name: str, role_fn: _RoleFn) -> None: ...
 def register_local_role(name: str, role_fn: _RoleFn) -> None: ...
 def role(
     role_name: str, language_module: _LanguageModule, lineno: int, reporter: Reporter
 ) -> tuple[_RoleFn | None, list[SystemMessage]]: ...
-def __getattr__(name: str) -> Incomplete: ...
+def set_implicit_options(role_fn: _RoleFn) -> None: ...
+def register_generic_role(canonical_name: str, node_class: Node) -> None: ...
+
+class GenericRole:
+    name: str
+    node_class: Node
+    def __init__(self, role_name: str, node_class: type[TextElement]) -> None: ...
```

Comment:
```suggestion
    def __init__(self, role_name: str, node_class: type[Node]) -> None: ...
```
Or change to TextElement above; it should be consistent.

Reply:
done. I've gone with `Node`, although looking at the code i suspect `TextElement` may be more accurate. thoughts?